src/Shared/HandyControl_Shared/Controls/Text/HighlightTextBlock.cs

```diff
@@ -0,0 +1,199 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Media;
+
+namespace HandyControl.Controls
+{
+    public class HighlightTextBlock : TextBlock
+    {
+        public static readonly DependencyProperty SourceTextProperty =
+            DependencyProperty.Register("SourceText", typeof(string), typeof(HighlightTextBlock), new PropertyMetadata(null, OnSourceTextChanged));
+        public static readonly DependencyProperty QueriesTextProperty =
+            DependencyProperty.Register("QueriesText", typeof(string), typeof(HighlightTextBlock), new PropertyMetadata(null, OnQueriesTextChanged));
+        public static readonly DependencyProperty HighlightBrushProperty =
+            DependencyProperty.Register("HighlightBrush", typeof(Brush), typeof(HighlightTextBlock), new PropertyMetadata(Brushes.Yellow));
+
+        private static void OnSourceTextChanged(DependencyObject d, DependencyPropertyChangedEventArgs e) =>
+            ((HighlightTextBlock) d).RefreshInlines();
+
+        private static void OnQueriesTextChanged(DependencyObject d, DependencyPropertyChangedEventArgs e) =>
+            ((HighlightTextBlock) d).RefreshInlines();
+
+        /// <summary>
+        /// Replace the original property Text for binding text.
+        /// </summary>
+        /// <remarks>
+        /// Don't use the <see cref="TextBlock.Text"/> property!
+        /// Because the <see cref="TextBlock.Text"/> has some unique behaviors
+        /// that is not needed at <see cref="HighlightTextBlock"/> at <see cref="TextBlock"/>,
+        /// which will cause some unexpected issue. 
+        /// </remarks>
+        public string SourceText
+        {
+            get => (string) GetValue(SourceTextProperty);
+            set => SetValue(SourceTextProperty, value);
+        }
+
+        /// <summary>
+        /// Gets or sets the text that need to be highlighted.
+        /// It can be an array of text separated by spaces.
+        /// </summary>
+        public string QueriesText
```

Comment:
@DingpingZhang  I think you mean **QueryText**, Right?

Reply:
What do you think about **HighlightText** instead of **QueryText**?

Reply:
Because the `QueriesText` is a set of strings that supports space separation, cf. line 74, I think the plural should be used here. Is there a grammatical error?